src/main/java/edu/cornell/kfs/fp/document/validation/impl/CuDisbursementVoucherDocumentPreRules.java

```diff
@@ -67,11 +69,30 @@ public boolean doPrompts(Document document) {
 
         // Handle custom pre-rules:
         preRulesOK &= this.checkCheckStubTextOverflow((DisbursementVoucherDocument) document);
+        
+        setIncomeClassNonReportableForForeignVendorWithNoTaxReviewRequired(document);
 
         return preRulesOK;
     }
 
-    /**
+	private void setIncomeClassNonReportableForForeignVendorWithNoTaxReviewRequired(Document document) {
+		DisbursementVoucherDocument dvDoc = (DisbursementVoucherDocument) document;
+		DisbursementVoucherPayeeDetail dvPayeeDetail = dvDoc.getDvPayeeDetail();
+		
+		String payeeTypeCode = dvPayeeDetail.getDisbursementVoucherPayeeTypeCode();
+		String paymentReasonCode = dvPayeeDetail.getDisbVchrPaymentReasonCode();
+		Integer vendorHeaderId = dvPayeeDetail.getDisbVchrVendorHeaderIdNumberAsInteger();
+
+		if (getCuDisbursementVoucherTaxService().isForeignVendorAndTaxReviewNotRequired(payeeTypeCode,paymentReasonCode, vendorHeaderId)) {
+			dvDoc.getDvNonResidentAlienTax().setIncomeClassCode(DisbursementVoucherConstants.NRA_TAX_INCOME_CLASS_NON_REPORTABLE);
```

Comment:
That is a good question. The income class field is set by the Tax reviewer when the document routes for tax review. In this case the doc will not route for tax review so the requirement was that a default value of 'N' is set in the income class field. The pre-rules are used for setting default values before business rules are applied so it seemed like the right place to do this. Bryan, do you have any suggestions? Thank you.